csharp/src/Ice/Connection.cs

```diff
@@ -102,6 +102,21 @@ public enum ConnectionClose
         GracefullyWithWait
     }
 
+    /// <summary>The state of an Ice connection</summary>
+    public enum ConnectionState : byte
+    {
+        /// <summary>The connection is being validated.</summary>
+        Validating = 0,
+        /// <summary>The connection is active and can send and receive messages.</summary>
+        Active,
+        /// <summary>The connection is being gracefully shutdown and waits for the peer to close its end of the
+        /// connection.</summary>
+        Closing,
+        /// <summary>The connection is closed and waits for potential dispatch to be finished before being destroyed
+        /// and detached from the observer.</summary>
```

Comment:
I would simplify this to "The connection is closed and eventually waits for potential dispatch to be finished before being destroyed.". I wonder if we should also add the "Destroyed" state... while not really useful to the connection internals, it's useful for observers to figure out when the connection is really completed (ie.: when CloseAsync returns).